src/subdomains/staking/application/services/staking.service.ts

```diff
@@ -47,14 +54,19 @@ export class StakingService {
     if (!asset) throw new NotFoundException('Asset not found');
 
     const existingStaking = await this.repository.findOne({ userId, asset, strategy });
-    if (existingStaking)
-      return StakingOutputDtoMapper.entityToDto(await this.authorize.authorize(userId, existingStaking.id));
+    if (existingStaking) {
+      const staking = await this.authorize.authorize(userId, existingStaking.id);
```

Comment:
Why do we need to authorize it again?

Reply:
we authorize only once, first time we just fetch existing staking to differentiate between staking not existing and user having no access to particular staking that already exists

Reply:
oh shoot... you right :)